### DISCUSSION TITLE
Code Refactoring

### DISCUSSION THREAD

**Discussion by ankitoscar** [Project-Easter/Flutter-UI#91] [Ideas]
Welcome aboard, contributors 🔥 :
Here you can discuss various ways in which we can make our code more modular and the places where we can do it.
While discussing your ideas, please mention me (@ankitoscar) or Mohit (@Mohitmadhav), so that we can reach out to you ASAP.

It'll include things like:
 - Improved State Management
 - Code Optimization
 - Reduce App size
 - Project Structuring (Removing or modifying files or folders), etc

**Comment by Mohitmadhav** [comment]
@keyurgit45 for the time being you can create a separate branch and try to execute null safety.
I'll assign it to you if I don't get further updates from the assignee.

Till then, lemme check for further bugs and issues.

**Reply by Mohitmadhav** [reply]
Or you can also try from scratch because I resolved the merge conflicts but there are still too many errors.

**Reply by Mohitmadhav** [reply]
Comment on the issue if you want to work on it

**Reply by keyurgit45** [reply]
If we want migrate to null safety all packages we use should support null safety but I doubt that packages we use in our project support it @Mohitmadhav

**Reply by Mohitmadhav** [reply]
Yep, that's why I wrote it here before, we need to sort that out.
Null safety is very important for the app.

**Reply by Mohitmadhav** [reply]
@RandyKdev you can also join this discussion if you can help us with this.

**Reply by keyurgit45** [reply]
> Yep, that's why I wrote it here before, we need to sort that out. Null safety is very important for the app.

ok let me see

**Reply by Mohitmadhav** [reply]
@keyurgit45 you can have a look at #103

**Reply by keyurgit45** [reply]
I checked that issue but I can't solve conflicts as I don't have write access. i have created new branch and trying to solve

**Reply by keyurgit45** [reply]
![image](https://user-images.githubusercontent.com/75253653/136565840-3730511f-36b7-46e4-80c9-63697ab0abf3.png)

**Reply by Mohitmadhav** [reply]
That is better because the guy who made the PR didn't have the updated main.
Okay then, first upgrade all packages before any migration.
I already upgraded many of them

**Reply by Mohitmadhav** [reply]
@keyurgit45 I'm literally irritated from this warning:

The plugin `geocoder` uses a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to see if this plugin supports the Android V2
embedding. Otherwise, consider removing it since a future release of Flutter will remove these deprecated APIs.
If you are plugin author, take a look at the docs for migrating the plugin to the V2 embedding:
https://flutter.dev/go/android-plugin-migration.

If you are able to get any solutions then look into it because it's not a bug or issue

**Reply by keyurgit45** [reply]
Okay!!

**Reply by keyurgit45** [reply]
https://pub.dev/packages/flutter_geocoder
can we use this package instead of geocoder? @Mohitmadhav 
just need to find alternatives for chips_choice, geocoder, lat long, qutoes.

**Reply by RandyKdev** [reply]
@keyurgit45 the above package does not support iOS (https://pub.dev/packages/flutter_geocoder/score).

**Reply by Mohitmadhav** [reply]
Right @RandyKdev ...also have a look at #103 to know the exact issue

**Reply by RandyKdev** [reply]
Ok, I will.

**Reply by RandyKdev** [reply]
@Mohitmadhav and @keyurgit45 Since the quotes package does not support null safety yet, it could be replaced entirely by using this api (http://quotes.rest/) directly. That is the api being used in the quotes package.

I am still searching for a fix or alternative for the geocoder package

**Reply by keyurgit45** [reply]
> @Mohitmadhav and @keyurgit45 Since the quotes package does not support null safety yet, it could be replaced entirely by using this api (http://quotes.rest/) directly. That is the api being used in the quotes package.
> 
> I am still searching for a fix or alternative for the geocoder package

Great!!

**Reply by Mohitmadhav** [reply]
@Phantom0001 you can also discuss here

**Reply by Mohitmadhav** [reply]
Till then @keyurgit45 you can take up other issues by tonight or tomorrow

**Reply by keyurgit45** [reply]
> Till then @keyurgit45 you can take up other issues by tonight or tomorrow

OK... I can't give that much time coz my exams are starting. But I Will try

**Reply by Mohitmadhav** [reply]
No problem, focus on your exams then. I notified you since you asked.
Best of Luck !!

**Comment by Phantom0001** [comment]
So what is out of date?

You want me to pull the main branch and check conflicts and re push?

**Reply by Mohitmadhav** [reply]
Yes @Phantom0001, I recently merged some PRs and made changes in upstream main.
Many things in your files were outdated with the main upstream branch

**Reply by Mohitmadhav** [reply]
So @Phantom0001 you can replace latlong with https://pub.dev/packages/latlong2. as @keyurgit45 said.
Actually, @HARSHBHUDOLIA and I used this before but got some conflicts at that time, don't remember exactly what were those.

**Comment by Mohitmadhav** [comment]
@aryang117  see if we can replace geocoder with some other package as we replaced latlong with latlong2 for null safety

**Reply by keyurgit45** [reply]
@Mohitmadhav I have created a separate branch and I replaced all packages to their null safety versions except chips choice and quotes. Also, we need to modify the getAddressfromLatLang function. and for chips choice, we can use https://pub.dev/packages/chips_choice_null_safety

**Reply by Mohitmadhav** [reply]
What did you replace Geocoder with exactly? According to that, we can make changes in getAddressFromLatLng

**Reply by keyurgit45** [reply]
> What did you replace Geocoder with exactly? According to that, we can make changes in getAddressFromLatLng

https://pub.dev/packages/geocode

**Reply by Mohitmadhav** [reply]
Cool, so after making the changes, you can take up #95

**Reply by keyurgit45** [reply]
> Cool, so after making the changes, you can take up #95 

Quotes is still remaining.. Which api should we use? @Mohitmadhav

**Reply by Mohitmadhav** [reply]
http://quotes.rest/
Try this out
As @RandyKdev suggested

**Reply by keyurgit45** [reply]
> http://quotes.rest/
> Try this out
> As @RandyKdev suggested

Can you provide the api key?

**Reply by Mohitmadhav** [reply]
https://theysaidso.com/api/
Here they showed how to use the endpoints

![image](https://user-images.githubusercontent.com/59333817/136824968-fd926009-4ec5-4534-a151-b0dac19fdfd1.png)
The free plan doesn't require the API key I guess (Only for Quote of the day)

**Reply by keyurgit45** [reply]
@Mohitmadhav Actually, that was quite problematic so I have used an open-source API. Will it be ok?
https://github.com/lukePeavey/quotable
![image](https://user-images.githubusercontent.com/75253653/136892351-a84a61ab-de9c-4952-a522-39ed203081c4.png)

**Reply by Mohitmadhav** [reply]
No problem, any free API will work for now

**Reply by aryang117** [reply]
> Cool, so after making the changes, you can take up #95

Since the geocoder issue will be resolved in the null safety fix, should I just make a PR (for #132 ) with just the `flutter_test` fix?

**Reply by Mohitmadhav** [reply]
Sure @aryang117

**Reply by keyurgit45** [reply]
@Mohitmadhav I have made the changes for null safety migration. Should I make PR or wait till @aryang117 pushes flutter_test fix?

**Reply by Mohitmadhav** [reply]
Yeah @keyurgit45

**Comment by Mohitmadhav** [comment]
https://pub.dev/packages/googleapis
